dev-packages/ovsx-client/src/ovsx-http-client.ts

```diff
@@ -34,28 +34,12 @@ export class OVSXHttpClient implements OVSXClient {
         protected requestService: RequestService
     ) { }
 
-    async search(searchOptions?: VSXSearchOptions): Promise<VSXSearchResult> {
-        try {
-            return await this.requestJson(this.buildUrl('api/-/search', searchOptions));
-        } catch (err) {
-            return {
-                error: err?.message || String(err),
-                offset: -1,
-                extensions: []
-            };
-        }
+    search(searchOptions?: VSXSearchOptions): Promise<VSXSearchResult> {
+        return this.requestJson(this.buildUrl('api/-/search', searchOptions));
```

Comment:
I believe this API has a consumer on the `browser` part of `@theia/vsx-registry`. I think it relies on `VSXSearchResult#error` being set - just throwing will lead to issues in the extension search widget.

Reply:
Yes, I know...we'll cross that bridge when we get to it.

Reply:
Ok, sure 👍